src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -578,12 +578,12 @@ public static void Move(String sourceDirName, String destDirName)
 
             StringComparison pathComparison = PathInternal.GetComparison();
 
-            if (String.Compare(sourcePath, destPath, pathComparison) == 0)
+            if (String.Equals(sourcePath, destPath, pathComparison))
```

Comment:
@justinvp agreed.  We should to a DotNET-bot pass to implement https://github.com/dotnet/corefx/issues/391.